### PR TITLE
HV-1822 add russian specific INN annotation

### PR DESCRIPTION
**Pull request by Boiarshinov**
Hi there!  
I tried to add russian specific `@INN` annotation.
I added annotation, validator, typedef, testcases, messages, etc.  
I've got problems with TCK runner, even after adding TestNG configuration like in this [guideline](http://hibernate.org/validator/contribute/). That's why i'm not shure that my build is stable. 
image

Also i've got problems with annotation processor module. Even without any changes (in the master branck) i've got a lot of failed tests. I think that they failed because of compilation difference. Here is one of the error messages (for `org.hibernate.validator.ap.ConstraintValidationProcessorTest#hibernateValidatorProvidedCustomConstraints` test):
```
Expected :DiagnosticExpectation [kind=ERROR, lineNumber=66], DiagnosticExpectation [kind=ERROR, lineNumber=67], DiagnosticExpectation [kind=ERROR, lineNumber=68], DiagnosticExpectation [kind=ERROR, lineNumber=69], DiagnosticExpectation [kind=ERROR, lineNumber=70], DiagnosticExpectation [kind=ERROR, lineNumber=71], DiagnosticExpectation [kind=ERROR, lineNumber=72], DiagnosticExpectation [kind=ERROR, lineNumber=73], DiagnosticExpectation [kind=ERROR, lineNumber=74], DiagnosticExpectation [kind=ERROR, lineNumber=75], DiagnosticExpectation [kind=ERROR, lineNumber=76], DiagnosticExpectation [kind=ERROR, lineNumber=77], DiagnosticExpectation [kind=ERROR, lineNumber=78], DiagnosticExpectation [kind=ERROR, lineNumber=79], DiagnosticExpectation [kind=ERROR, lineNumber=80], DiagnosticExpectation [kind=ERROR, lineNumber=81], DiagnosticExpectation [kind=ERROR, lineNumber=82], DiagnosticExpectation [kind=ERROR, lineNumber=83], DiagnosticExpectation [kind=ERROR, lineNumber=84]
Actual   :DiagnosticExpectation [kind=ERROR, lineNumber=12], DiagnosticExpectation [kind=ERROR, lineNumber=13], DiagnosticExpectation [kind=ERROR, lineNumber=14], DiagnosticExpectation [kind=ERROR, lineNumber=15], DiagnosticExpectation [kind=ERROR, lineNumber=16], DiagnosticExpectation [kind=ERROR, lineNumber=17], DiagnosticExpectation [kind=ERROR, lineNumber=18], DiagnosticExpectation [kind=ERROR, lineNumber=19], DiagnosticExpectation [kind=ERROR, lineNumber=20], DiagnosticExpectation [kind=ERROR, lineNumber=21], DiagnosticExpectation [kind=ERROR, lineNumber=22], DiagnosticExpectation [kind=ERROR, lineNumber=23], DiagnosticExpectation [kind=ERROR, lineNumber=24], DiagnosticExpectation [kind=ERROR, lineNumber=25], DiagnosticExpectation [kind=ERROR, lineNumber=26], DiagnosticExpectation [kind=ERROR, lineNumber=27], DiagnosticExpectation [kind=ERROR, lineNumber=28], DiagnosticExpectation [kind=ERROR, lineNumber=29], DiagnosticExpectation [kind=ERROR, lineNumber=30], DiagnosticExpectation [kind=ERROR, lineNumber=31], DiagnosticExpectation [kind=ERROR, lineNumber=33], DiagnosticExpectation [kind=ERROR, lineNumber=40], DiagnosticExpectation [kind=ERROR, lineNumber=41], DiagnosticExpectation [kind=ERROR, lineNumber=42], DiagnosticExpectation [kind=ERROR, lineNumber=43], DiagnosticExpectation [kind=ERROR, lineNumber=44], DiagnosticExpectation [kind=ERROR, lineNumber=45], DiagnosticExpectation [kind=ERROR, lineNumber=46], DiagnosticExpectation [kind=ERROR, lineNumber=47], DiagnosticExpectation [kind=ERROR, lineNumber=48], DiagnosticExpectation [kind=ERROR, lineNumber=49], DiagnosticExpectation [kind=ERROR, lineNumber=50], DiagnosticExpectation [kind=ERROR, lineNumber=51], DiagnosticExpectation [kind=ERROR, lineNumber=52], DiagnosticExpectation [kind=ERROR, lineNumber=53], DiagnosticExpectation [kind=ERROR, lineNumber=54], DiagnosticExpectation [kind=ERROR, lineNumber=55], DiagnosticExpectation [kind=ERROR, lineNumber=56], DiagnosticExpectation [kind=ERROR, lineNumber=59], DiagnosticExpectation [kind=ERROR, lineNumber=60], DiagnosticExpectation [kind=ERROR, lineNumber=66], DiagnosticExpectation [kind=ERROR, lineNumber=67], DiagnosticExpectation [kind=ERROR, lineNumber=68], DiagnosticExpectation [kind=ERROR, lineNumber=69], DiagnosticExpectation [kind=ERROR, lineNumber=70], DiagnosticExpectation [kind=ERROR, lineNumber=71], DiagnosticExpectation [kind=ERROR, lineNumber=72], DiagnosticExpectation [kind=ERROR, lineNumber=73], DiagnosticExpectation [kind=ERROR, lineNumber=74], DiagnosticExpectation [kind=ERROR, lineNumber=75], DiagnosticExpectation [kind=ERROR, lineNumber=76], DiagnosticExpectation [kind=ERROR, lineNumber=77], DiagnosticExpectation [kind=ERROR, lineNumber=78], DiagnosticExpectation [kind=ERROR, lineNumber=79], DiagnosticExpectation [kind=ERROR, lineNumber=80], DiagnosticExpectation [kind=ERROR, lineNumber=81], DiagnosticExpectation [kind=ERROR, lineNumber=82], DiagnosticExpectation [kind=ERROR, lineNumber=83], DiagnosticExpectation [kind=ERROR, lineNumber=84]
```
I'm newbie with annotation processors and absolutly get stuck with it.
Maybe this is the compiler issue (or maybe not). So for your info: I used JDK 1.8.0_241

### CODE REVIEW

**Comment by Boiarshinov** [comment]
Wow! There is a lot of changes in UTF-8 symbols. I didn't see these changes in my IDE. 
I will try to fix it in the evening.

**Comment by gsmet** [comment]
ok to test

File: engine/src/test/java/org/hibernate/validator/test/cfg/ProgrammaticConstraintDefinitionsTest.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertNoViolations;
 import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertThat;
 import static org.hibernate.validator.testutil.ConstraintViolationAssert.violationOf;
+import static org.hibernate.validator.testutils.ValidatorUtil.getConfiguration;
```
Comment (gsmet) [inline engine/src/test/java/org/hibernate/validator/test/cfg/ProgrammaticConstraintDefinitionsTest.java:12]: Checkstyle is complaining about this unused import.

**Review by gsmet — CHANGES_REQUESTED** [review]
Nice work!

Could you fix the small Checkstyle issue and squash all the commits?

I'll have a closer look after the Christmas break.

File: engine/src/test/java/org/hibernate/validator/test/cfg/ProgrammaticConstraintDefinitionsTest.java (hunk 1 above)
Comment (Boiarshinov) [reply engine/src/test/java/org/hibernate/validator/test/cfg/ProgrammaticConstraintDefinitionsTest.java]: Fixed

**Comment by Boiarshinov** [comment]
> Could you fix the small Checkstyle issue and squash all the commits?  

All done.  

Thank you for your time and attention. Have a great Christmas Eve!

**Comment by gsmet** [comment]
I amended your commit to fix a small typo and compile the all digits pattern in the validator to avoid recompiling it for each execution.

I will merge it when CI is green and backport it to 6.2 too.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks again for this nice (and very clean) contribution!

**Comment by Boiarshinov** [comment]
Thank you too. 
This is my first contribution to opensource. I'm very happy now!

**Comment by gsmet** [comment]
Congratulations! Very nice work for a first time, very thorough.

FYI, I released HV 6.2.0.Final and 7.0.0.Final including your contribution today and they will be available in Maven Central tomorrow.

**Pull request merged**